lib/active_record_doctor/detectors/extraneous_indexes.rb

```diff
@@ -41,9 +41,10 @@ def subindexes_of_multi_column_indexes
             next if maximal_indexes.include?(index)
 
             replacement_indexes = maximal_indexes.select do |maximum_index|
-              cover?(maximum_index, index)
+              compatible_options?(maximum_index, index) && cover?(maximum_index, index)
             end.map(&:name).sort
 
+            next if replacement_indexes.empty?
```

Comment:
These changes must be reverted and the `#cover?` method must be patched instead. Let me explain the thinking behind this algorithm.

Indexes form a partially ordered set - you can visualize them using directed graphs or Hasse diagrams. Indexes are vertices in a directed graph and there's an edge from index `a` to index `b` if `a` can replace `b`. In the code we call it `#cover?` so if `cover?(a, b)` returns `true` then `a` covers (can replace) `b`.

The problem is `#cover?` doesn't check for operator classes. Currently, it takes uniqueness into account. We must modify `#cover?` to take operator classes into account so that when it receives two indexes with different types or different operator classes then it always returns `false`. After doing that, the algorithm should do just fine without any changes to `subindexes_of_multi_column_indexes`.